user_id.py

```diff
@@ -1,33 +1,23 @@
-"""Use a given name string, a family name string, and a random integer
-between 100 and 999 to create a user_id.
+"""Use two strings, 'given_name' and 'family_name', to create a hash
+value as the user_id.
 """
-import random
 import re
+import hashlib
 import name
 
 def is_valid(user_id_str):
     """Take in a string containing a user ID. Return True if it is
-    valid user ID. Otherwise, return False.
+    valid. Otherwise, return False.
     """
-    if re.search("[a-zA-Z]+[0-9][0-9][0-9]", user_id_str):
-        return True
-    return False
+    return True if re.search("[a-zA-Z0-9]{56}", user_id_str) else False
```

Comment:
you have any idea how much trouble we'd have saved ourselves if we had this in our own code?